llvm/test/MC/AMDGPU/gfx12_asm_sopp.s

```diff
@@ -271,7 +272,9 @@ s_cbranch_execnz 0x1234
 // GFX12: s_cbranch_execnz 4660 ; encoding: [0x34,0x12,0xa6,0xbf]
 
 s_barrier
-// GFX12: s_barrier ; encoding: [0x00,0x00,0xbd,0xbf]
+// GFX12-ERR: :[[@LINE-1]]:{{[0-9]+}}: error: instruction not supported on this GPU
```

Comment:
Please add this case to gfx12_unsupported.s instead. No need to change the RUN lines in this test.